### ISSUE TITLE
Error with matplotlib backend when rendering Japanese whitespace to PNG

### ISSUE THREAD

**Issue by ggand0** [mozman/ezdxf#1033]
**Describe the bug**
I encounter the error `ValueError: 'vertices' must be 2D with shape (N, 2), but your input has shape (1, 3)` when attempting to convert a DXF file containing text with Japanese whitespace using the Matplotlib backend. The issue is observed with a free Japanese .shx font provided [here](https://domisan.sakura.ne.jp/article/cadfont/cadfont.html).

Summarized error trace:
```
Traceback (most recent call last):
  File "<my_proj_location>/debug.py", line 35, in <module>
    Frontend(ctx, out).draw_layout(msp,layout_properties=layout_properties, finalize=True)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 237, in draw_layout
    self.draw_entities(
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 271, in draw_entities
    _draw_entities(self, self.ctx, entities, filter_func=filter_func)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 810, in _draw_entities
    frontend.draw_entity(entity, properties)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 298, in draw_entity
    draw_method(entity, properties)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 379, in draw_mtext_entity
    self.draw_complex_mtext(mtext, properties)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/frontend.py", line 396, in draw_complex_mtext
    complex_mtext_renderer(self.ctx, self.designer, mtext, properties)
  ...
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/tools/text_layout.py", line 487, in render
    self.renderer.render(  # type: ignore
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/mtext_complex.py", line 175, in render
    self.backend.draw_text(self.text, t, self.properties, self.cap_height)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/designer.py", line 400, in draw_text
    self._draw_path(text_path, properties)
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/designer.py", line 280, in _draw_path
    self.backend.draw_path(path, self.get_backend_properties(properties))
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/addons/drawing/matplotlib.py", line 159, in draw_path
    mpl_path = to_matplotlib_path([path])
  File "<my_venv_location>/lib/python3.9/site-packages/ezdxf/npshapes.py", line 494, in to_matplotlib_path
    return Path(np.concatenate(vertices), codes)
  File "<my_venv_location>/lib/python3.9/site-packages/matplotlib/path.py", line 130, in __init__
    _api.check_shape((None, 2), vertices=vertices)
  File "<my_venv_location>/lib/python3.9/site-packages/matplotlib/_api/__init__.py", line 161, in check_shape
    raise ValueError(
ValueError: 'vertices' must be 2D with shape (N, 2), but your input has shape (1, 3)

```

**To Reproduce**
1. Create a .dxf file with AutoCAD and add a single text object "あ　" (note that the whitespace is a Japanese one) and save as `test.dxf`. 
2. Create an environment with Python 3.9.1, ezdxf==1.1.4, matplotlib==3.8.3
3. Download the font file `azomix.shx` and place it under `fonts` directory
4. Specify `support_dirs = "./fonts"` in `ezdxf.ini` and run `ezdxf --fonts`
5. Run the following snippet
```
import ezdxf
import matplotlib.pyplot as plt
from ezdxf.addons.drawing import RenderContext, Frontend
from ezdxf.addons.drawing.matplotlib import MatplotlibBackend
from ezdxf.addons.drawing.properties import LayoutProperties

input_dxf_path = "<your_dxf_path>"
output_png_path = "<desired_png_path>"
dpi = 300

doc = ezdxf.readfile(input_dxf_path)
msp = doc.modelspace()
fig, ax = plt.subplots()

for style in doc.styles:
    style.dxf.font = "azomix.shx"

fig = plt.figure()
ctx = RenderContext(doc)
layout_properties = LayoutProperties.from_layout(msp)
layout_properties.set_colors(bg='#FFFFFF')
ax = fig.add_axes([0, 0, 1, 1])
out = MatplotlibBackend(ax)

Frontend(ctx, out).draw_layout(msp,layout_properties=layout_properties, finalize=True)

fig.savefig(output_png_path, dpi=dpi)
plt.close(fig)
```

**Confidential Data**
No confidential data is required to reproduce this error. I confirmed that I encounter this error with a newly created dxf file.

**Expected behavior**
MatplotlibBackend handles Japanese whitespace properly so that user can convert DXF files with Japanese text to PNG files without errors.

**Screenshots**
I also have attached a zip including test.dxf and relevant codes.
[ezdxf-debug.zip](https://github.com/mozman/ezdxf/files/14340982/ezdxf-debug.zip)
<img width="1552" alt="ss0" src="https://github.com/mozman/ezdxf/assets/1145044/2c049880-aaef-4e95-be86-6db1fe8ead11">

**Comment by mozman** [comment]
Hi!

Thank you for the detailed bug report and the link to first really free SHX and LFF fonts, which I can include in the repository!

The issue seems to be fixed in the current beta version v1.2.0b3:

BUT the file `test.dxf` uses a different font `aehalf.shx` which is replaced on my system:

![image](https://github.com/mozman/ezdxf/assets/4230966/17527128-e48c-40c5-a464-3de150add768)


Replacement font, I guess `arialuni.ttf`:

![image](https://github.com/mozman/ezdxf/assets/4230966/302550a7-85fd-4be4-a26a-8f9724b7976c)

While the SVG output `out.svg` seems to be correct - for some unknown reasons 😒:

![out](https://github.com/mozman/ezdxf/assets/4230966/e2560a38-eb27-4bdb-b241-3e93acdcb607)


When I edit the DXF file and replace `aehalt.shx` by `azomix.shx` (which is installed now) and add some `xxx` to see if whitespace is rendered properly, I get this:

![image](https://github.com/mozman/ezdxf/assets/4230966/3e8bca2d-7ba1-43e2-a022-822b57c34c73)

and for `kst32b.shx` I get this:

![image](https://github.com/mozman/ezdxf/assets/4230966/a624c3c6-0810-47d1-8042-5c7d36cd9a5a)

I don't fully understand what happens here (I have written all the related code!), but it seems to be fixed with version 1.2.0.

**Comment by ggand0** [comment]
Thank you for your quick reply and letting me know about the fix. Yes, I chose that font randomly for the test.dxf, but originally I got this error with another DXF file that using `txt.shx`, which also had Japanese whitespace. I can confirm that using `ezdxf==v1.2.0b2` seems to fix this issue. I ran the same code, and both the edited test.dxf (which now includes "あ　xxx" with `aehalf.shx`) and the other DXF file were rendered to PNG without any errors.
![out](https://github.com/mozman/ezdxf/assets/1145044/4cb0a0c4-2fa0-45f3-8e7e-2bdafa8c7b45)
